op-chain-ops/addresses/roles.go

```diff
@@ -0,0 +1,30 @@
+package addresses
+
+import "github.com/ethereum/go-ethereum/common"
+
+type L1Roles struct {
+	SuperchainRoles
+	OpChainRoles
+}
+
+type SuperchainRoles struct {
+	SuperchainProxyAdminOwner common.Address
+}
+
+type OpChainRoles struct {
+	OpChainCoreRoles
+	OpChainFaultProofsRoles
+}
+
+type OpChainCoreRoles struct {
+	SystemConfigOwner      common.Address
+	OpChainProxyAdminOwner common.Address
+	Guardian               common.Address
+	Proposer               common.Address
```

Comment:
Proposer, like Challenger, is a more of an FP role.

Reply:
Oh, good point/observation. 

The [proposer](https://specs.optimism.io/glossary.html?highlight=proposer#proposer) is a role that exists both pre-fault proofs, and in the permissioned dispute game, but not in the permissionless dispute game.

Pre-fault proofs the `proposer` role was defined in the `L2OutputOracle`, which I would also say is technically a component of the legacy proof system.

Reply:
Ok in that case I'm inclined to leave it as part of the `OpChainCoreRoles` struct since it is a role that every chain gets, regardless of whether they deploy the `FaultProofsContracts`

Reply:
Ok you've changed my mind on this. I think its better to optimize for the standard case, not the legacy/non-standard configuration. Moved that role to `OpChainFaultProofsRoles`. It can still be accessed in the same way regardless on the parent struct (i.e. `OpChainRoles.Proposer`). Done in https://github.com/ethereum-optimism/optimism/pull/15506/commits/3cbb774a090d6db4833400e6867a98bbe3f4d952